File: hw 3/fractional.py
import random

from number import *

class Fractional(Number):
    def __init__(self):
        self.number = 0
        self.denominator = 0

    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.number = int(strArray[i])
        self.denominator = int(strArray[i+1])
        i += 2
        return i

    def Print(self):
        print("It is fraction: Numerator = {}, Denominator = {}. Convert to real number {}".format(self.number, self.denominator, self.ConvertToRealNumber()))
        pass

    def Write(self, ostream):
        ostream.write("It is fraction: Numerator = {}, Denominator = {}. Convert to real number {}".format(self.number, self.denominator, self.ConvertToRealNumber()))
        pass

    def InRnd(self):
        self.number = random.uniform(1, 2000)
        self.denominator = random.uniform(1, 2000)

    def ConvertToRealNumber(self):
        return self.number / self.denominator
        pass

File: hw_2/fractional.cpp
//------------------------------------------------------------------------------
// complex.cpp - содержит процедуру ввода параметров
// для уже созданного комплексного числа
//------------------------------------------------------------------------------

#include "fractional.h"
#include <fstream>

//------------------------------------------------------------------------------
Fractional::Fractional() : numerator(0), denominator(1) {}

Fractional::~Fractional() {}
// Ввод дроби из файла
void Fractional::In(std::ifstream &ifst) {
    ifst >> numerator >> denominator;
}

// Случайный ввод комплексного числа
void Fractional::InRnd() {
    numerator = Random();
    denominator = Random();
}

//------------------------------------------------------------------------------
// Вывод дроби в форматируемый поток
void Fractional::Out(std::ofstream &ofst) {
    ofst << "It is fraction: Numerator = " << numerator
         << ", Denominator = " << denominator
         << ". Convert to real number " << ConvertToRealNumber() << "\n";
}

//------------------------------------------------------------------------------
// Вычисление действительного представления дроби
double Fractional::ConvertToRealNumber() {
    return ((double) numerator / denominator);
}
